src/remover.rs

```diff
@@ -1,81 +1,66 @@
-use crate::field_finder::{get_field, TomlValue};
 use std::{io::Error, io::ErrorKind};
+
+use anyhow::{bail, Context, Result};
 use toml_edit::{Array, ArrayOfTables, Document, InlineTable, Table};
 
-pub fn handle_remove(field: String, doc: &mut Document) -> Result<(), Error> {
+use crate::field_finder::{get_field, DoInsert, TomlValue};
+
+pub fn handle_remove(field: &str, doc: &mut Document) -> Result<()> {
     let mut path_split = field
         .split('/')
         .map(|s| s.to_string())
         .collect::<Vec<String>>();
 
-    let last_field = match path_split.pop() {
-        Some(last_field) => last_field,
-        None => return Err(Error::new(ErrorKind::Other, "Path is empty")),
-    };
+    let last_field = path_split.pop().context("path is empty")?;
 
-    let insert_if_not_exists = false;
-    let field = match get_field(&path_split, &last_field, insert_if_not_exists, doc) {
+    let field = match get_field(&path_split, &last_field, DoInsert::No, doc) {
         Ok(field) => field,
-        Err(e) => {
-            if e.kind() == ErrorKind::NotFound {
+        Err(e) => match e.downcast::<Error>() {
```

Comment:
This is the one place where using `anyhow` bites us. `anyhow` is great for errors in binaries where we just report errors to users, whereas regular errors (also see `thiserror`) are better for internal stuff where you actually want to change behavior based on which error you get.

So it might make sense to use a new error enum + `thiserror` just for `field_finder.rs`. I personally would avoid `std::io::Error` here since thats designed for io errors